### ISSUE TITLE
If npm run scripts fail, it should stop maven

### ISSUE THREAD

**Issue by SeabassNg** [eirslett/frontend-maven-plugin#412]
In our current configuration, when we run npm scripts and it fails, the maven build moves on. There should be a configuration somewhere to stop the build if the scripts fail.

**Comment by eirslett** [comment]
Check the exit status of your failing scripts? Maven will stop if the script returns a non-zero value.

**Comment by rradillen** [comment]
I'm having the same issue and it looks like the exit code is non-zero. But there may be a layer in between that swallows that exit code. Do you have a way to see what exit code maven actually receives?

**Comment by mriehema** [comment]
Interesting. I will test it in our setup with a sass error (mvn -> npm -> grunt -> sass).

**Comment by mriehema** [comment]
Well, works here.

```
[INFO] Running "sass:build" (sass) task
[INFO] >> Error: Undefined variable: "$screen-xs-max".
...
[INFO] Aborted due to warnings.
...
[ERROR] npm ERR! foo-theme@1.0.0 postinstall: `grunt build`
[ERROR] npm ERR! Exit status 6
...
[ERROR] npm ERR! @coremedia/common@1.0.0 postinstall: `node ./lib/scripts/postinstall.js`
[ERROR] npm ERR! Exit status 1
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:22 min
[INFO] Finished at: 2016-09-22T11:09:52+02:00
[INFO] Final Memory: 73M/961M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.0:npm (install-node-modules) on project frontend: Failed to run task: 'npm install' failed. (error code 1) -> [Help 1]
```

Like @eirslett said, check your npm scripts to return correct exit codes. As ou can see, we have a layer between grunt and npm too, which checkes the exit codes too.

The code looks like (based on shelljs):
```
      if (shell.exec('npm install').code !== 0) {
        shell.echo("Error: Installation of theme " + theme.name + " failed. ");
        shell.exit(1);
      }
```

**Comment by brandones** [comment]
I ran into this and fixed it by upgrading from 1.4 to 1.11.3. I expect that this issue can be closed.
